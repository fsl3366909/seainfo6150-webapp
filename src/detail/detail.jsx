import React, { useState, useEffect } from 'react';
import { Switch, Route, Link } from "react-router-dom";
import './index.css';
const Detail = (props) => {
    const [data, setData] = useState({
        image: { _url: '' },
        name: '',
        price: '',
        type: ''
    });
    console.log(props.match.params.uid)
    useEffect(() => {
        fetch('https://run.mocky.io/v3/7a21bf9b-1ddb-42ca-9583-472e215c44e6', {
            method: 'GET',
            headers: {
                'content-type': 'application/x-www-form-urlencoded'
            },
        }).then(response => response.text())
            .then((response) => {
                // console.log(response.replace(/},/g, "}},"));
                var txt = response.replace(/},/g, "}},");/* 处理数组 */
                txt = txt.replace(/[\r\n]/g, "").replace(/[ ]/g, "");
                txt = '[' + txt.split('}},}')[0] + '}}}]'
                console.log(JSON.parse(txt))
                let data = JSON.parse(txt)
                setData(Object.values(data[0])[props.match.params.uid])
                console.log(Object.values(data[0])[props.match.params.uid])
            }).catch((err) => {
                console.log(err)
            }); // connection errors
    }, []);
    return (
        <div className='detail'>
            <h1>Detail</h1>{/* {data.image._url} */}
            <img className='pngblock' src={data.image._url} alt="" />
            <div className='contxt'>
                <p className='txt'>name:{data.name}</p>
                <p className='txt'>price:{data.price}</p>
                <p className='txt'>type:{data.type}</p>
            </div>
            <br />
            <br />
            <br />
            <br />
        </div>
    )
}

export default Detail
