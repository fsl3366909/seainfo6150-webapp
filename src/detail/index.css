/* 产品列表 */
#product {
  width: 1200px;
  display: block;
  margin: auto;
}
#product .btn-group {
  display: flex;
}
#product .btn-group .tit {
  width: 100px;
  color: #999;
  text-align: right;
}
#product .btn-group .btn {
  flex-grow: 1;
  padding-left: 30px;
  color: #333;
}
#product ul {
  display: grid;
  margin: auto;
  grid-template-columns: 33% 33% 33%;
}
#product li {
  width: 100%;
  text-align: center;
}
#product li .pro {
  height: 500px;
  overflow: hidden;
  margin: 20px;
  border: 1px solid #ccc;
}
#product li .png {
  width: 90%;
  text-align: center;
  height: 300px;
  display: block;
  margin: auto;
  overflow: hidden;
  position: relative;
}
#product li .png img {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  display: block;
  width: 100%;
  margin: auto;
}
#product li .txt {
  font-size: 24px;
  line-height: 40px;
  text-align: left;
  padding-left: 30px;
}
#product .defaultbtn {
  display: block;
  margin: 30px auto;
  width: 160px;
  height: 30px;
  font-size: 18px;
  border: 1px solid #333;
  color: #333;
  opacity: 0.8;
  line-height: 30px;
  background: #f4f4f4;
}
#product .defaultbtn:hover {
  background: #000;
  color: #fff;
}
