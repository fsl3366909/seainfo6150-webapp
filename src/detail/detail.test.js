import React from 'react';
import detail from './detail';

describe('ArticleList tests', () => {
    const article = {
        "name": "bmw-530",
        "type": "sedan",
        "price": "40000",
        "image": {
            "_url": "https://img5.autotimes.com.cn/news/2019/04/0426_142700657113.jpg",
            "_height": "619",
            "_width": "1100"
        },
    };
    it('renders correctly', () => {
        const { container } = render(<detail article={article} />);
        expect(container).toMatchSnapshot();
    });
});