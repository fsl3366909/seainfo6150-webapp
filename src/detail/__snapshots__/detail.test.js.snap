// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <detail
    article="[object Object]"
  />
</div>
`;
